### PR TITLE
[aws-datastore] Improve list-scattering routines in test coe

### PR DESCRIPTION
**Pull request by jamesonwilliams**
A couple of tests rely on the ability to create a unique, randomly
ordered version of an input list.

Properly obtaining a scattered list has a couple of nuances. So,
this routine is moved to a utility class.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: testutils/src/main/java/com/amplifyframework/testutils/Scattered.java [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    @Nullable
+    public static <T> List<T> list(@Nullable List<T> original) {
+        if (original == null || original.size() <= 1) {
+            return Immutable.of(original);
+        }
+        SecureRandom random = new SecureRandom();
+        List<T> scattered = new ArrayList<>(original);
+        while (scattered.equals(original)) {
+            //noinspection ComparatorMethodParameterNotUsed Intentional; result is random
+            Collections.sort(scattered, (one, two) -> random.nextInt());
```
Comment (jpignata) [inline testutils/src/main/java/com/amplifyframework/testutils/Scattered.java:62]: Thought this looked familiar: https://github.com/aws-amplify/amplify-android/pull/446. From that pull:

> I'm assuming the sort with a random integer was not guaranteeing an even distribution. Any reason not to use `Collections.shuffle()` whose docs claims it does?

File: testutils/src/main/java/com/amplifyframework/testutils/Scattered.java [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+     *
+     *   1. A null, empty, or singleton list cannot be scattered; just return an immutable
+     *      copy, immediately.
+     *
+     *   2. For lists that have more than 1 element, we can sort the list using a comparator
+     *      that decides at random whether one element is larger than another. A SecureRandom
+     *      is used for this, to reduce the likelihood that the same ordering will be generated
+     *      across multiple runs.
+     *
+     *   3. For lists with small amounts of elements, the likelihood of attaining a unique
```
Comment (jpignata) [inline testutils/src/main/java/com/amplifyframework/testutils/Scattered.java:45]: Distant cousin of bogosort. ;)

File: testutils/src/main/java/com/amplifyframework/testutils/Scattered.java (hunk 1 above)
Comment (jamesonwilliams) [reply testutils/src/main/java/com/amplifyframework/testutils/Scattered.java:62]: Oh lol!!!! I didn't know about that one. Thanks.

https://docs.oracle.com/javase/7/docs/api/java/util/Collections.html#shuffle(java.util.List)

File: testutils/src/main/java/com/amplifyframework/testutils/Scattered.java (hunk 2 above)
Comment (jamesonwilliams) [reply testutils/src/main/java/com/amplifyframework/testutils/Scattered.java]: Haha yea, true! "bogoshuffle"

Sounds like a dance.

**Pull request merged**